src/rtc/mosaic_geobursts.py

```diff
@@ -0,0 +1,341 @@
+'''
+A module to mosaic Sentinel-1 geobursts from RTC workflow
+'''
+
+import os
+
+import numpy as np
+from osgeo import osr, gdal
+
+def check_reprojection(list_rtc_imagery: list, list_nlooks: list,
+                             geogrid_mosaic) -> bool:
+    '''
+    Check if the reprojection is required to mosaic the input list of the rasters
+
+    Parameters:
+    -----------
+    list_rtc_imagery: list
+        path to the geocoded RTC images to mosaic
+    list_nlooks: list
+        path to the nlooks raster that corresponds to list_rtc
+
+    Returns:
+    flag_requires_reprojection: bool
+        True if reprojection is necessary to mosaic `list_rtc_imagery`;
+        False if the images are aligned, so that no reprojection is necessary.
+        None if the mosaicking is not possible.
+    '''
+
+    # Accepted error in the coordinates as floating number. Used when checking the snapping.
+    maxerr_coord = 1.0e-6
+
+    # Return value after this reprojection requirement check.
+    # It will turn to `True` if reprojection is necessary,
+    # or `None` if mosaicking is not possible.
+    flag_requires_reprojection = False
+
+    # Check the number of the raster files in list_rtc_imagery and list_nlooks
+    num_rtc_imagery = len(list_rtc_imagery)
+    num_nlooks = len(list_nlooks)
+    if num_rtc_imagery != num_nlooks:
+        print (f'# RTC ({num_rtc_imagery}) and # nlooks ({num_nlooks}) does not match.')
+        flag_requires_reprojection = None
```

Comment:
There's no reason to have this `if` and and a huge `else`. Just return flag_requires_reprojection here if it enters the if, and then you don't need the else statement